trade-module/src/main/java/io/edpn/backend/trade/adapter/persistence/repository/MybatisLocateSingleHopTradeRouteRepository.java

```diff
@@ -0,0 +1,118 @@
+package io.edpn.backend.trade.adapter.persistence.repository;
+
+import io.edpn.backend.trade.adapter.persistence.entity.MybatisSingleHopEntity;
+import io.edpn.backend.trade.adapter.persistence.entity.MybatisStationEntity;
+import io.edpn.backend.trade.adapter.persistence.entity.MybatisValidatedCommodityEntity;
+import io.edpn.backend.trade.adapter.persistence.filter.MybatisLocateSingleHopTradeFilter;
+import org.apache.ibatis.annotations.One;
+import org.apache.ibatis.annotations.Result;
+import org.apache.ibatis.annotations.Results;
+import org.apache.ibatis.annotations.Select;
+
+import java.util.List;
+
+public interface MybatisLocateSingleHopTradeRouteRepository {
+    
+    @Select("""
+            <script>
+            WITH stations_in_range AS (SELECT station.id,
+                                              system_id,
+                                              arrival_distance
+                                       FROM public.station
+                                                INNER JOIN system ON station.system_id = system.id
+                                       WHERE st_3ddwithin(st_makepoint(#{xCoordinate}, #{yCoordinate}, #{zCoordinate}), coordinates_geom, 300)
+                                        <if test='!includeFleetCarriers'>
+                                         AND station.name NOT SIMILAR TO '[A-Za-z0-9]{3}-[A-Za-z0-9]{3}'
+                                        </if>
+                                        <if test='maxLandingPadSize == "LARGE"'>AND max_landing_pad_size = 'LARGE'</if>
+                                        <if test='maxLandingPadSize == "MEDIUM"'>AND max_landing_pad_size IN ('MEDIUM', 'LARGE')</if>
```

Comment:
Not really something that should (or reasonably could) be addressed in a drive-by fashion as part of an unrelated PR, but something worth considering: `IN` is not exactly performant either. Perhaps it would be a good to have numeric values for landing pad sizes, so that we can utilize greater/smaller than? One solution would be an "enum table" of the landing pad sizes, with names and numeric values, and the numeric values doubling as primary keys.

Reply:
Its a low cardinality table. The cost of this is < 63 ms. This seems like overkill for a performance problem we're not having. Our latest trade data in a while is to the scale of approx 2Gb, if we were working with larger data sets I would apply this. But until we see a need for it I don't see a gain to making the data less readable.